### ISSUE TITLE
Premature termination when running on guava with N=1000

### ISSUE THREAD

**Issue by khatchad** [ponder-lab/Logging-Level-Evolution-Plugin#166]
N=1000, use only config as a category.
```
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.FINE,"Starting {0}.",service)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is Closeables.logger.log(Level.WARNING,"Suppressing exception thrown when closing " + closeable,suppressed)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.FINE,"Started {0} in {1}.",new Object[]{service,stopwatch})
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.WARNING,"ServiceManager configured with no services.  Is your application configured properly?",new EmptyServiceManagerWarning())
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkIfBlock
INFO: We meet a logging wrapping: 
if (logger.isLoggable(Level.SEVERE)) {
  logger.log(Level.SEVERE,message(context),exception);
}

Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.SEVERE,message,throwable)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.WARNING,"Error loading regex compiler, falling back to next option",e)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is log.log(Level.SEVERE,"UnsafeAtomicHelper is broken!",thrownUnsafeFailure)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.FINE,"Service {0} has terminated. Previous state was: {1}",new Object[]{service,from})
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(SEVERE,String.format(Locale.ROOT,"Caught an exception in %s.  Shutting down.",t),e)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.warning("Cannot read directory " + directory)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is log.log(Level.SEVERE,"SafeAtomicHelper is broken!",thrownReflectionFailure)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.SEVERE,"Service " + service + " has failed in the "+ from+ " state.",failure)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is log.log(Level.SEVERE,"SafeAtomicHelper is broken!",thrownAtomicReferenceFieldUpdaterFailure)
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.WARNING,"ignoring weigher specified without maximumWeight")
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer getRejuvenatedLogLevel
INFO: Use log category: do not consider config.
Mar 14, 2019 4:01:50 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer analyzeLogInvs
INFO: Do action: CONVERT_TO_FINEST! The changed log expression is logger.log(Level.SEVERE,"Detected potential deadlock",e)
...
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
Mar 14, 2019 4:16:48 PM edu.cuny.hunter.log.core.analysis.LogAnalyzer checkEnoughData
INFO: The DOI values are all same or no DOI values. Cannot get valid results.
...
!ENTRY org.eclipse.ltk.ui.refactoring 4 10000 2019-03-14 16:38:12.153
!MESSAGE Internal Error
!STACK 0
java.lang.reflect.InvocationTargetException
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:401)
	at org.eclipse.jface.wizard.WizardDialog.run(WizardDialog.java:977)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizard.createChange(RefactoringWizard.java:665)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizard.computeUserInputSuccessorPage(RefactoringWizard.java:506)
	at org.eclipse.ltk.ui.refactoring.UserInputWizardPage.computeSuccessorPage(UserInputWizardPage.java:77)
	at org.eclipse.ltk.ui.refactoring.UserInputWizardPage.getNextPage(UserInputWizardPage.java:113)
	at org.eclipse.jface.wizard.WizardDialog.nextPressed(WizardDialog.java:854)
	at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:410)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:622)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:84)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5663)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1386)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4931)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4472)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:822)
	at org.eclipse.jface.window.Window.open(Window.java:798)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation$1.run(RefactoringWizardOpenOperation.java:192)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:72)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.run(RefactoringWizardOpenOperation.java:212)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.run(RefactoringWizardOpenOperation.java:126)
	at org.eclipse.jdt.internal.ui.refactoring.actions.RefactoringStarter.activate(RefactoringStarter.java:41)
	at edu.cuny.hunter.log.ui.wizards.LogWizard.startRefactoring(LogWizard.java:221)
	at edu.cuny.hunter.log.ui.handlers.LogHandler.execute(LogHandler.java:57)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:294)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:96)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:320)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:254)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:164)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:497)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:490)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5663)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1386)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4931)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4472)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:636)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1501)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1474)
Caused by: java.lang.NullPointerException
	at edu.cuny.hunter.log.core.refactorings.LogRejuvenatingProcessor.createChange(LogRejuvenatingProcessor.java:316)
	at org.eclipse.ltk.core.refactoring.participants.ProcessorBasedRefactoring.createChange(ProcessorBasedRefactoring.java:291)
	at org.eclipse.ltk.core.refactoring.CreateChangeOperation.run(CreateChangeOperation.java:125)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2295)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2322)
	at org.eclipse.ltk.internal.ui.refactoring.WorkbenchRunnableAdapter.run(WorkbenchRunnableAdapter.java:89)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)
Root exception:
java.lang.NullPointerException
	at edu.cuny.hunter.log.core.refactorings.LogRejuvenatingProcessor.createChange(LogRejuvenatingProcessor.java:316)
	at org.eclipse.ltk.core.refactoring.participants.ProcessorBasedRefactoring.createChange(ProcessorBasedRefactoring.java:291)
	at org.eclipse.ltk.core.refactoring.CreateChangeOperation.run(CreateChangeOperation.java:125)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2295)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2322)
	at org.eclipse.ltk.internal.ui.refactoring.WorkbenchRunnableAdapter.run(WorkbenchRunnableAdapter.java:89)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)
```

**Comment by khatchad** [comment]
The problem doesn't happen if I pick the setting for CONFIG/WARNING/SEVERE.

**Comment by khatchad** [comment]
But now I am getting that the DOI values are all the same except for `ServiceManager.java`.

**Comment by saledouble** [comment]
I never saw this error before. Let me check.

**Comment by saledouble** [comment]
I could see the mylyn-git plugin bump DOI values for the methods changed in `guava-testlib`, but their all DOI values are all 0. Looks weird.

**Comment by saledouble** [comment]
Breakpoint at:
https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/blob/158a02c3f5adc644cecfd7977189023289f238fe/edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/refactorings/LogRejuvenatingProcessor.java#L278

![image](https://user-images.githubusercontent.com/10117031/54464925-275e7100-474f-11e9-980e-91ef0243f374.png)

**Comment by saledouble** [comment]
I got the same issue while running the developer's UI.

**Comment by khatchad** [comment]
Why would a log invocation not have an enclosing method?

**Comment by saledouble** [comment]
We have an issue here: https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/issues/75

**Comment by khatchad** [comment]
So, basically, this is caused by a logging statement that is not within a method. I see that your fix is to not do the transformation for such logging statements, however, currently, such logging statements should count as failures by our analysis. Questions:

1. Is that currently the case?
1. What is the root cause of this particular problem? Is it a field initializer or static initializer?

**Comment by saledouble** [comment]
I already counted it as a `CURRENTLY_NOT_HANDLED` failure. 

1. Yes.
2. It is static initializer:

https://github.com/google/guava/blob/88e6fb86f5317bdfd2e8a78899334e9f0ba16987/guava/src/com/google/common/util/concurrent/AbstractFuture.java#L131-L174

**Comment by saledouble** [comment]
These two log invocations have no enclosing methods:

https://github.com/google/guava/blob/88e6fb86f5317bdfd2e8a78899334e9f0ba16987/guava/src/com/google/common/util/concurrent/AbstractFuture.java#L170-L172

**Comment by khatchad** [comment]
> I already counted it as a `CURRENTLY_NOT_HANDLED` failure. 

Then why are we trying to transformed a failed logging statement? The fix should be to check if there is an error status with the logging statement right?

> 1. Yes.
> 2. It is static initializer:

> https://github.com/google/guava/blob/88e6fb86f5317bdfd2e8a78899334e9f0ba16987/guava/src/com/google/common/util/concurrent/AbstractFuture.java#L131-L174

Please run this through the JavaModel or ASTView plugin. How hard is it to get the static initializer object? It should have a common super type with `IMethod` (in fact, it does, it's called [`IMember`](https://help.eclipse.org/neon/topic/org.eclipse.jdt.doc.isv/reference/api/org/eclipse/jdt/core/IMember.html). I feel that the `getEnclosingMethod()` should be renamed to `getEnclosingMember()` and it should return an `IMember` and that this will partially fix #75. I think it's OK to not deal with fields but I don't think that static initializers are all that different.

**Comment by saledouble** [comment]
>Then why are we trying to transformed a failed logging statement? The fix should be to check if there is an error status with the logging statement right?

I fixed at first, then I added a failure. Now, I realize that the fix becomes useless because of the added failure. We are not trying to transform a failed logging statement. Right. I will move the fix into the correct place.

>Please run this through the JavaModel or ASTView plugin. 

OK.
